base/file.jl

```diff
@@ -344,6 +358,9 @@ tempdir()
     tempname()
 
 Generate a unique temporary file path.
+Warning: this can lead to race conditions if another process obtains the same
```

Comment:
You could make this a Documenter-style warning:
```
!!! warning

    This can lead to race conditions if another process obtains the same
    file name and creates the file before you are able to.
    Using [`mktemp()`](@ref) is recommended instead.
```